### ISSUE TITLE
FluentValidation and Middleware ordering

### ISSUE THREAD

**Issue by BrianVallelunga** [JasperFx/wolverine#506]
Is the FluentValidation middleware supposed to run before custom middleware if configured in this way?

``` f#
app.MapWolverineEndpoints(fun opts ->
        opts.AddMultitenacyMiddleware()
        opts.UseFluentValidationProblemDetailMiddleware())
```

My custom MultitenancyMiddleware just has a single `Before` method that sets the `bus.TenantId` based on a header passed in the request. It works fine, but runs after the fluent validation middleware.  I would expect with the above ordering that the multitenancy one would run first.

This is problematic because the validator needs to know the tenant to work against. I'm injecting `IMessageBus` into a custom validator and its `TenantId` is null. After the validator runs, then the middleware runs.

Thanks

**Comment by jeremydmiller** [comment]
There's a topological sort that happens for all the codegen frames. Had to say w/o seeing your code, but sometimes it will turn things around in ways you don't really want. If this is for an HTTP endpoint (and it looks like it is), I think the multi-tenancy improvements in v1.7 coming out within the next couple hours will fix this all up for you.

I'm going to be overconfident and close this for now.

**Comment by BrianVallelunga** [comment]
I hate to say it , but 1.8.0 does not fix this issue. It did indeed fix #569 but `TenantId` is still not set when a FluentValidator is run. I have a simple validator that injects IMessageContext and its `TenantId` is always null. I checked this with `IBus` as well and the result is the same. This is for an HTTP endpoint.

The configuration looks like this:

```f#
app.MapWolverineEndpoints(fun opts ->
    opts.TenantId.IsRequestHeaderValue("Tenant")
    opts.UseFluentValidationProblemDetailMiddleware())
```
